packages/core/src/modules/credentials/v2/CredentialExchangeRecord.ts

```diff
@@ -0,0 +1,86 @@
+import type { TagsBase } from '../../../storage/BaseRecord'
+import type { AutoAcceptCredential } from '../CredentialAutoAcceptType'
+import type { CredentialProtocolVersion } from '../CredentialProtocolVersion'
+import type { CredentialState } from '../CredentialState'
+import type { CredentialRole } from './CredentialRole'
+
+import { BaseRecord } from '../../../storage/BaseRecord'
+
+export interface CredentialRecordTags {
+  threadId: string
+  protocolVersion: CredentialProtocolVersion
+  state: CredentialState
+  connectionId?: string
+}
+
+export enum CredentialRecordType {
+  INDY,
+  W3C,
```

Comment:
I would assign string values to the enum values as we will be storing the values. If you don't assign a value it will be auto-assigned a number based on the order of declaration

So in this case INDY = 0, W3C = 1, if we later add an entry at the top of the array, INDY = 1, W3C = 2. 
 

Also we normally use PascalCase for enums

```suggestion
  Indy = "Indy",
  W3c = "W3c",
```